src/hazelcore/ZExp.re

```diff
@@ -512,7 +610,10 @@ and mk_inconsistent_zoperand = (u_gen, zoperand) =>
   | InjZ(InHole(TypeInconsistent, _), _, _)
   | CaseZE(StandardErrStatus(InHole(TypeInconsistent, _)), _, _)
   | CaseZR(StandardErrStatus(InHole(TypeInconsistent, _)), _, _)
-  | ApPaletteZ(InHole(TypeInconsistent, _), _, _, _) => (zoperand, u_gen)
+  | ApPaletteZ(InHole(TypeInconsistent, _), _, _, _)
```

Comment:
can you go back and make ApPalette the last case in all the pattern sequences involving it, instead of putting Subscript at the end? that will make future merges easier.